comp/process/apiserver/apiserver_test.go

```diff
@@ -43,13 +50,54 @@ func TestLifecycle(t *testing.T) {
 		fetchonlyimpl.MockModule(),
 	))
 
-	assert.Eventually(t, func() bool {
-		res, err := http.Get("http://localhost:6162/config")
-		if err != nil {
-			return false
-		}
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		req, err := http.NewRequest("GET", "http://localhost:43424/agent/status", nil)
+		require.NoError(c, err)
+		util.CreateAndSetAuthToken(pkgconfigsetup.Datadog())
+		req.Header.Set("Authorization", "Bearer "+util.GetAuthToken())
+		res, err := util.GetClient(false).Do(req)
+		require.NoError(c, err)
 		defer res.Body.Close()
+		assert.Equal(c, http.StatusOK, res.StatusCode)
+	}, 5*time.Second, time.Second)
+}
+
+func TestPostAuthentication(t *testing.T) {
+	_ = fxutil.Test[Component](t, fx.Options(
+		Module(),
+		core.MockBundle(),
+		fx.Replace(config.MockParams{Overrides: map[string]interface{}{
+			"process_config.cmd_port": 43424,
+		}}),
+		workloadmetafx.Module(workloadmeta.NewParams()),
+		fx.Supply(
+			status.Params{
+				PythonVersionGetFunc: func() string { return "n/a" },
+			},
+		),
+		taggerfx.Module(tagger.Params{
+			UseFakeTagger: true,
+		}),
+		statusimpl.Module(),
+		settingsimpl.MockModule(),
+		fetchonlyimpl.MockModule(),
```

Comment:
Same as in the `TestLifecycle` test
```suggestion
		createandfetchimpl.Module(),
```

Reply:
Done